daemons/compute/lib-cpp/CMakeLists.txt

```diff
@@ -42,45 +51,55 @@ add_custom_command(
         "${hw_proto}"
       DEPENDS "${hw_proto}")
 
-# Include generated *.pb.h files
-include_directories("${CMAKE_CURRENT_BINARY_DIR}")
+#
+# nnfdm library: libnnfdm.so for shared library and libnnfdm.a for static
+#
+if(BUILD_SHARED_LIBS)
+  add_library(nnfdm SHARED ${generated_srcs} client.cc client.h)
+else()
+  add_library(nnfdm STATIC ${generated_srcs} client.cc client.h)
+endif()
+
+#
+# Compile with position independant code if we are set to run as a shared library
+#
+set_target_properties(nnfdm PROPERTIES POSITION_INDEPENDANT_CODE ${BUILD_SHARED_LIBS})
+
+# Include generated *.pb.h files privately as these headers are for the library's
+# internal implementation.
+#
+target_include_directories(nnfdm PRIVATE "${CMAKE_CURRENT_BINARY_DIR}")
 
-# hw_grpc_proto
-add_library(hw_grpc_proto
-  ${hw_grpc_srcs}
-  ${hw_grpc_hdrs}
-  ${hw_proto_srcs}
-  ${hw_proto_hdrs})
-target_link_libraries(hw_grpc_proto
-  ${_REFLECTION}
-  ${_GRPC_GRPCPP}
-  ${_PROTOBUF_LIBPROTOBUF})
+# Noted: _REFLECTION, _GRPC_GRPCPP, and _PROTOBUF_LIBPROTOBUF are
+#        set by common.cmake
+#
+target_link_libraries(nnfdm
+    PUBLIC
+    ${_REFLECTION}
+    ${_GRPC_GRPCPP}
+    ${_PROTOBUF_LIBPROTOBUF}
+)
 
-# Targets greeter_[async_](client|server)
-# foreach(_target
-#   client)
-#   add_executable(${_target} "${_target}.cc")
-#   target_link_libraries(${_target}
-#     hw_grpc_proto
-#     ${_REFLECTION}
-#     ${_GRPC_GRPCPP}
-#     ${_PROTOBUF_LIBPROTOBUF})
-# endforeach()
+#
+# Export a cmake configuration file for our library so downstream users
+# may (more easily) configure this library in to their cmake-based builds.
+#
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+  "${PROJECT_SOURCE_DIR}/nnfdm-config.cmake.in"
+  "${PROJECT_BINARY_DIR}/nnfdm-config.cmake"
+  INSTALL_DESTINATION share/cmake/nnfdm)
 
-include(GNUInstallDirs)
-add_library(mylib SHARED client.cc)
-link_libraries()
-target_link_libraries(mylib
-  hw_grpc_proto
-  ${_REFLECTION}
-  ${_GRPC_GRPCPP}
-  ${_PROTOBUF_LIBPROTOBUF})
-set_target_properties(mylib
-    PROPERTIES VERSION ${PROJECT_VERSION}
-    SOVERSION 1
-    PUBLIC_HEADER api/mylib.h)
-# configure_file(mylib.pc.in mylib.pc @ONLY)
-target_include_directories(mylib PRIVATE .)
-install(TARGETS mylib
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+# Generate install of when running 'make install':
```

Comment:
This comment is missing something.
Generate install of <what?> when running `make install`:

Reply:
Ah, good question.  I probably should have left this comment out.  But the subsequent 'install' cmake directories will be used by cmake to generate the install `make` target so that running "make install" will yield the following  files.  The piece I wanted to capture was to depict where the installation would go and what the layout would be. 

Perhaps I should just remove the entire comment since it may all be derived by the four install directives below?

Reply:
I'd prefer you leave the comment section in, maybe remove the word `of`?
# Generate install when running 'make install':